orangecontrib/text/tests/test_preprocess.py

```diff
@@ -1,381 +1,226 @@
+        self.assertIn('porter', str(stemmer).lower())
+
+    def test_call(self):
+        word = "Testing"
+        tokens = ["Testing", "tokenized", "Sentence"]
+        stemmer = preprocess.PorterStemmer()
+        self.assertEqual(stemmer(word), self.stemmer(word))
+        self.assertEqual(stemmer(tokens),
+                         [self.stemmer(token) for token in tokens])
+
+    def test_function(self):
+        stemmer = preprocess.BaseNormalizer()
+        stemmer.normalizer = lambda x: x[:-1]
+        self.assertEqual(stemmer.normalize('token'), 'toke')
+
+    def test_snowball(self):
+        stemmer = preprocess.SnowballStemmer()
+        stemmer.language = 'french'
+        token = 'voudrais'
+        self.assertEqual(stemmer(token), nltk.SnowballStemmer(language='french').stem(token))
+
+    def test_call_with_bad_input(self):
+        stemmer = preprocess.PorterStemmer()
+        self.assertRaises(TypeError, stemmer, 10)
+
+    def test_lookup_normalize(self):
+        dln = preprocess.DictionaryLookupNormalizer(dictionary={'aka': 'also known as'})
+        self.assertEqual(dln.normalize('aka'), 'also known as')
+
+
+class FilteringTests(unittest.TestCase):
+    def test_call(self):
+        class DigitsFilter(preprocess.BaseTokenFilter):
+            def check(self, token):
+                return not token.isdigit()
+
+        df = DigitsFilter()
+
+        self.assertEqual(df([]), [])
+        self.assertEqual(df(['a', '1']), ['a'])
+        self.assertEqual(df([['a', '1']]), [['a']])
+
+    def test_stopwords(self):
+        filter = preprocess.StopwordsFilter('english')
+
+        self.assertFalse(filter.check('a'))
+        self.assertTrue(filter.check('filter'))
+
+    def test_lexicon(self):
+        filter = preprocess.LexiconFilter(['filter'])
+        self.assertFalse(filter.check('false'))
+        self.assertTrue(filter.check('filter'))
+
+
+class BaseTokenizerTests(unittest.TestCase):
+    def test_call(self):
+        class DashTokenizer(preprocess.BaseTokenizer):
+            @classmethod
+            def tokenize(cls, string):
+                return string.split('-')
+
+        tokenizer = DashTokenizer()
+        self.assertEqual(list(tokenizer('dashed-sentence')), ['dashed', 'sentence'])
+        self.assertEqual(list(tokenizer(['1-2-3', '-'])), [['1', '2', '3'], ['', '']])
+
+        self.assertRaises(TypeError, tokenizer, 1)
+
+    def test_tokenizer_instance(self):
+        class WhitespaceTokenizer(preprocess.BaseTokenizer):
+            tokenizer = nltk.WhitespaceTokenizer()
+            name = 'whitespace'
+
+        tokenizer = WhitespaceTokenizer()
+
+        sent = "Test \t tokenizer"
+        self.assertEqual(tokenizer.tokenize(sent),
+                         nltk.WhitespaceTokenizer().tokenize(sent))
+
+    def test_options(self):
+        tokenizer = preprocess.RegexpTokenizer(pattern='\w')
+        self.assertTrue(hasattr(tokenizer, 'pattern'))
+        tokenizer.pattern = r'\w+'
```

Comment:
What's the intuition for setting this first to `\w` and the reseting to `\w+`. To check if setters work?